pkg/karpenter/karpenter.go

```diff
@@ -12,28 +12,37 @@ import (
 const (
 	// DefaultKarpenterNamespace default namespace for Karpenter
 	DefaultKarpenterNamespace = "karpenter"
+	// DefaultKarpenterServiceAccountName is the name of the service account which is needed for Karpenter
+	DefaultKarpenterServiceAccountName = "karpenter"
 
-	karpenterHelmRepo         = "https://charts.karpenter.sh"
-	karpenterHelmChartName    = "karpenter/karpenter"
-	karpenterReleaseName      = "karpenter"
-	controllerClusterName     = "controller.clusterName"
-	controllerClusterEndpoint = "controller.clusterEndpoint"
-	createServiceAccount      = "serviceAccount.create"
-	addDefaultProvisioner     = "defaultProvisioner.create"
+	karpenterHelmRepo      = "https://charts.karpenter.sh"
+	karpenterHelmChartName = "karpenter/karpenter"
+	karpenterReleaseName   = "karpenter"
+	controller             = "controller"
+	clusterName            = "clusterName"
+	clusterEndpoint        = "clusterEndpoint"
+	serviceAccount         = "serviceAccount"
+	defaultProvisioner     = "defaultProvisioner"
+	create                 = "create"
+)
+
+const (
+	addDefaultProvisionerDefaultValue = false
 )
 
 // Options contains values which Karpenter uses to configure the installation.
 type Options struct {
-	HelmInstaller         providers.HelmInstaller
-	Namespace             string
-	ClusterName           string
-	AddDefaultProvisioner bool
-	CreateServiceAccount  bool
-	ClusterEndpoint       string
-	Version               string
+	HelmInstaller        providers.HelmInstaller
+	Namespace            string
+	ClusterName          string
+	CreateServiceAccount bool
+	ClusterEndpoint      string
+	Version              string
 }
 
 // InstallKarpenter defines a functionality to install Karpenter.
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -generate
+//counterfeiter:generate -o fakes/fake_karpenter_installer.go . InstallKarpenter
 type InstallKarpenter interface {
```

Comment:
there's some repetition here too with `karpenter.InstallKarpeneter`, which could be omitted

Reply:
This one has a history from the previous PR. Naming this wasn't super easy. :D So we decided to keep this one around. :) Check it out here: https://github.com/weaveworks/eksctl/pull/4480

Reply:
However, if you have a better name PLEASE SHARE!! :D

Reply:
`karpenter.ChartInstaller`?

Reply:
or `helm.KarpenterInstaller`

Reply:
ChartInstaller! I like that one! I already have a package called `helm` so I can't use that. ChartInstaller isn't bad.

Reply:
it fits! go for it